[hotfix] Fix drop database not working in terminal

MarigWeizhi opened the pull request:
<!--
Thanks for sending a pull request!

Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://amoro.apache.org/how-to-contribute/
  2. If the PR is related to an issue in https://github.com/apache/amoro/issues, add '[AMORO-XXXX]' in your PR title, e.g., '[AMORO-XXXX] Your PR title ...'.
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][AMORO-XXXX] Your PR title ...'.
-->

## Why are the changes needed?
<!--
Please clarify why the changes are needed. For instance,
  1. If you add a feature, you can talk about its use case.
  2. If you fix a bug, you can clarify why it is a bug.
  3. Use Fix/Resolve/Close #{ISSUE_NUMBER} to link this PR to its related issue
-->

Close #xxx.

## Brief change log
<!--
Clearly describe the changes made in modules, classes, methods, etc.
-->

- Added the `dropNamespace` to support cascade or non-cascade in `SparkUnifiedCatalogBase` for spark 3.3
- spark 3.2 will use false by default

## How was this patch tested?

- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? ( no)
- If yes, how is the feature documented? (not documented)

huyuanfeng2018 reviewed (commented):
Should we add a test to guard against future regression?

MarigWeizhi commented:
> Should we add a test to guard against future regression?

Added test, PTAL

huyuanfeng2018 commented on amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java:
```diff
@@ -86,7 +89,20 @@ public UnboundFunction loadFunction(Identifier ident) throws NoSuchFunctionExcep
+      throw new NoSuchNamespaceException(namespace);
+    }
+    List<TableIDWithFormat> tables = unifiedCatalog.listTables(database);
+    if (!tables.isEmpty() && !cascade) {
```
Why not drop all tables first and then drop the database?

I think should this logic be implemented directly by the parent (spark-common).
We already have concrete logic in  `SparkUnifiedCatalogBase.dropNamespace`.
so，remove Override is enough.

WDYT?

zhoujinsong replied:
I agreed that we may reuse some implementation in the parent class. But I think it is dangerous to delete all tables by default in Spark 3.2 implementation. So wee may:

- Add a method in the base class to support dropping namespace cascade or not.
- Drop namespace by default with false for Spark 3.2

huyuanfeng2018 reviewed (commented):
Thanks for the PR!

MarigWeizhi commented:
Thanks for your review @zhoujinsong @huyuanfeng2018   
I updated the code, I implemented the method to support cascade or not in base instead of spark 3.2.

huyuanfeng2018 commented on amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java:
```diff
@@ -85,8 +85,8 @@ public UnboundFunction loadFunction(Identifier ident) throws NoSuchFunctionExcep
    */
   @Override
   public boolean dropNamespace(String[] namespace, boolean cascade)
```
Just delete this method, WDYT?

MarigWeizhi replied:
deleted

huyuanfeng2018 reviewed (commented):
just a nit comment

huyuanfeng2018 reviewed (approved):
LGTM.

zhoujinsong commented:
Thanks for the contribution! @MarigWeizhi 
Thanks for the review! @huyuanfeng2018
